libsigner/src/events.rs

```diff
@@ -209,27 +217,45 @@ impl EventReceiver for StackerDBEventReceiver {
                     &request.method(),
                 )));
             }
-            if request.url() != "/stackerdb_chunks" {
-                let url = request.url().to_string();
+            if request.url() == "/stackerdb_chunks" {
+                debug!("Got stackerdb_chunks event");
+                let mut body = String::new();
+                request
+                    .as_reader()
+                    .read_to_string(&mut body)
+                    .expect("failed to read body");
 
-                info!(
-                    "[{:?}] next_event got request with unexpected url {}, return OK so other side doesn't keep sending this",
-                    event_receiver.local_addr,
-                    request.url()
-                );
+                let event: StackerDBChunksEvent =
+                    serde_json::from_slice(body.as_bytes()).map_err(|e| {
+                        EventError::Deserialize(format!("Could not decode body to JSON: {:?}", &e))
+                    })?;
+
+                if !contract_ids.contains(&event.contract_id) {
+                    info!(
+                        "[{:?}] next_event got event from an unexpected contract id {}, return OK so other side doesn't keep sending this",
+                        event_receiver.local_addr,
+                        event.contract_id
+                    );
+                    request
+                        .respond(HttpResponse::empty(200u16))
+                        .expect("response failed");
+                    return Err(EventError::UnrecognizedStackerDBContract(event.contract_id));
+                }
 
                 request
                     .respond(HttpResponse::empty(200u16))
                     .expect("response failed");
-                Err(EventError::UnrecognizedEvent(url))
-            } else {
+
+                Ok(SignerEvent::StackerDB(event))
+            } else if request.url() == "/proposal_response" {
```

Comment:
Same thing here -- as soon as the body is read into the string, return 200.

Reply:
Done!